tests/test_reflection.py

```diff
@@ -0,0 +1,99 @@
+import pytest
+from sqlalchemy import MetaData, Table
+from sqlalchemy.schema import CreateTable
+
+from rs_sqla_test_utils import models
+
+
+def cleaned(statement):
+    stripped_lines = [line.strip() for line in statement.split('\n')]
+    joined = '\n'.join([line for line in stripped_lines if line])
+    return joined
+
+
+def table_to_ddl(session, table):
+    return str(CreateTable(table)
+               .compile(session.bind))
+
+
+models_and_ddls = [
+    (models.ReflectionDistKey, """
+    CREATE TABLE reflection_distkey (
+        col1 INTEGER NOT NULL,
+        col2 INTEGER,
+        PRIMARY KEY (col1)
+    ) DISTSTYLE KEY DISTKEY (col1)
+    """),
+    (models.ReflectionSortKey, """
+    CREATE TABLE reflection_sortkey (
+        col1 INTEGER NOT NULL,
+        col2 INTEGER,
+        PRIMARY KEY (col1)
+    ) DISTSTYLE EVEN SORTKEY (col1, col2)
+    """),
+    (models.ReflectionInterleavedSortKey, """
+    CREATE TABLE reflection_interleaved_sortkey (
+        col1 INTEGER NOT NULL,
+        col2 INTEGER,
+        PRIMARY KEY (col1)
+    ) DISTSTYLE EVEN INTERLEAVED SORTKEY (col1, col2)
+    """),
+    (models.ReflectionUniqueConstraint, """
+    CREATE TABLE reflection_unique_constraint (
+        col1 INTEGER NOT NULL,
+        col2 INTEGER,
+        PRIMARY KEY (col1),
+        UNIQUE (col1, col2)
+    ) DISTSTYLE EVEN
+    """),
+    (models.ReflectionPrimaryKeyConstraint, """
+    CREATE TABLE reflection_pk_constraint (
+        col1 INTEGER NOT NULL,
+        col2 INTEGER NOT NULL,
+        PRIMARY KEY (col1, col2)
+    ) DISTSTYLE EVEN
+    """),
+    (models.ReflectionForeignKeyConstraint, """
+    CREATE TABLE reflection_fk_constraint (
+        col1 INTEGER NOT NULL,
+        col2 INTEGER,
+        PRIMARY KEY (col1),
+        FOREIGN KEY(col1) REFERENCES reflection_unique_constraint (col1)
+    ) DISTSTYLE EVEN
+    """),
+    (models.ReflectionDefaultValue, """
+    CREATE TABLE reflection_default_value (
+        col1 INTEGER NOT NULL,
+        col2 INTEGER DEFAULT 5,
+        PRIMARY KEY (col1)
+    ) DISTSTYLE EVEN
+    """),
+    (models.ReflectionIdentity, """
+    CREATE TABLE reflection_identity (
+        col1 INTEGER NOT NULL,
+        col2 INTEGER IDENTITY(1,3),
+        PRIMARY KEY (col1)
+    ) DISTSTYLE EVEN
+    """),
+    # # The column name with quotes currently fails to be reflected as pk
+    # (models.ReflectionDelimitedIdentifiers, '''
+    # CREATE TABLE "group" (
+    #     "this ""is it""" INTEGER NOT NULL,
+    #     PRIMARY KEY ("this ""is it""")
+    # ) DISTSTYLE EVEN
+    # '''),
```

Comment:
Okay update the FK code as well (replace the part where you split on ','  with a regex). Then add this and an FK test as xfails. We can merge this and try and get an upstream fix in